honeybee_3dm/grid.py

```diff
@@ -0,0 +1,77 @@
+"""Import Rhino surfaces as Honeybee grid objects from a Rhino3DM file."""
+import warnings
+# The Rhino3dm library provides the ability to access content of a Rhino3dm
+# file from outside of Rhino
+import rhino3dm
+
+# Importing Honeybee dependencies
+from honeybee_radiance.sensorgrid import SensorGrid
+from honeybee.typing import clean_and_id_string, clean_string
+from ladybug_geometry.geometry3d import Face3D, Mesh3D
+
+# Importing dependencies from Honeybee-3dm package
+from .togeometry import brep2d_to_face3d, mesh_to_face3d, mesh_to_mesh3d
+from .helper import filter_objects_by_layer
+
+
+# TODO: expose an option to change the target layer name
+# TODO: Grid size and offset will have to provided using the config file
+
+
+def import_grids(rhino3dm_file, tolerance, grid_size=1, grid_offset=0):
+    """Creates Honeybee grids from a rhino3dm file.
+
+    This function assumes all the grid objects are under a layer named ``grid``.
+
+    Args:
+        rhino3dm_file: The rhino file from which Honeybee grids will be created.
+        tolerance: A rhino3dm tolerance object. Tolerance set in the rhino file.
+        grid_size: Grid size. Defaults to 1.
+        grid_offset: Grid offset from the grid geometry. Defaults to 0.
+
+    Returns:
+        A list of Honeybee grids.
+    """
+
+    # Objects from rhino file
+    try:
+        grid_objs = filter_objects_by_layer(rhino3dm_file, 'grid')
+        warnings.warn(
+            'Only objects on layer "grid" will be used to create grids.')
+    except ValueError:
+        # no layer named grid
+        return []
+
+    hb_grids = []
+
+    for obj in grid_objs:
+        try:
+            geo = obj.Geometry
+            # If it's a Brep
+            if isinstance(geo, rhino3dm.Brep):
+                face3d = brep2d_to_face3d(geo, tolerance)[0]
+                mesh3d = face3d.mesh_grid(grid_size, grid_size, grid_offset)
+
+            # If it's a Mesh
+            elif isinstance(geo, rhino3dm.Mesh):
+                # If it's a Mesh with only one face
+                if len(geo.Faces) == 1:
+                    face3d = mesh_to_face3d(geo)[0]
+                    mesh3d = face3d.mesh_grid(grid_size, grid_size, grid_offset)
```

Comment:
I think we should deal with all the meshes the same. Why are you making an exception for meshes with one face?

Reply:
For the left hand side mesh (1 face) we can create grid points. For the right hand side meshes (More faces) we use mesh face centroids.
![why mesh](https://user-images.githubusercontent.com/20086924/99155836-9b21f100-2689-11eb-8cb1-f4f677eb8bc3.jpg)

Reply:
I think depending on how we want to present this to user it might be confusing. I imagine similar to Ladybug we want to use the cenetriod of the Mesh as is so the user can mesh the geometries before saving them and have full control on where the sensors will end up being. We don't want to change that on your side I guess.

Reply:
We're not changing it. We're still doing that. My point is, if have mesh with 1 mesh face then we need go divide that into a grid of meshes first. That's why I am treating a mesh with one face separately.

Reply:
>  we need go divide that into a grid of meshes first.

That's exactly my point. The input mesh should stay intact unless we remesh all the input meshes. This means that we are leaving the meshing control fully in the hands of the user when they create the meshes in Rhino. That being said if we find a better solution using Blender then we can start taking advantage of that and provide re-meshing all the input meshes as an option.

Reply:
@mostaphaRoudsari, I propose that we talk about this. Might take 10-15 mins max. It is most likely that we both want to achieve the same thing.

Reply:
I think I got your point. You want that we don't divide a one face mesh into grids and leave that to the users. I will make this change.